hazelcast/src/main/java/com/hazelcast/sql/impl/schema/map/sample/MapSampleMetadataResolver.java

```diff
@@ -48,40 +50,42 @@ private MapSampleMetadataResolver() {
     /**
      * Resolves the metadata associated with the given key-value sample.
      *
-     * @param ss Serialization service.
+     * @param ss     Serialization service.
      * @param target Target to be analyzed.
-     * @param key Whether passed target is key or value.
+     * @param key    Whether passed target is key or value.
      * @return Sample metadata.
      * @throws QueryException If metadata cannot be resolved.
      */
     public static MapSampleMetadata resolve(
-        InternalSerializationService ss,
-        JetMapMetadataResolver jetMapMetadataResolver,
-        Object target,
-        boolean key
+            InternalSerializationService ss,
+            JetMapMetadataResolver jetMapMetadataResolver,
+            Object target,
+            boolean key
     ) {
         try {
-            // Convert Portable object to Data to have consistent object fields irrespectively of map's InMemoryFormat.
-            if (target instanceof Portable) {
-                target = ss.toData(target);
-            }
-
             if (target instanceof Data) {
                 Data data = (Data) target;
 
                 if (data.isPortable()) {
                     return resolvePortable(ss.getPortableContext().lookupClassDefinition(data), key, jetMapMetadataResolver);
+                } else if (data.isCompact()) {
+                    CompactInternalGenericRecord internalGenericRecord =
+                            (CompactInternalGenericRecord) ss.readAsInternalGenericRecord(data);
+                    return resolveCompact(internalGenericRecord.getSchema(), key, jetMapMetadataResolver);
                 } else if (data.isJson()) {
                     throw new UnsupportedOperationException("To query JSON objects, "
                             + "you have to define the data model using CREATE MAPPING.");
                 } else {
                     return resolveClass(ss.toObject(data).getClass(), key, jetMapMetadataResolver);
                 }
+            } else if (target instanceof Portable) {
+                // Convert Portable object to Data to have consistent object fields irrespectively of map's InMemoryFormat.
+                Data data = ss.toData(target);
+                return resolvePortable(ss.getPortableContext().lookupClassDefinition(data), key, jetMapMetadataResolver);
             } else if (target instanceof PortableGenericRecord) {
-                // We get PortableGenericRecord here when in-memory format is OBJECT and
-                // the cluster does not have PortableFactory configuration for this Portable.
-                // PortableGetter can extract fields PortableGenericRecord.
                 return resolvePortable(((PortableGenericRecord) target).getClassDefinition(), key, jetMapMetadataResolver);
+            } else if (target instanceof CompactInternalGenericRecord) {
```

Comment:
Why it is `PortableGenericRecord` above (instead of `PortableInternalGenericRecord`), but `CompactInternalGenericRecord` here (instead of `CompactGenericRecord`)?

Reply:
Looks like a leftover. I will put `CompactGenericRecord` here